src/twisted/internet/defer.py

```diff
@@ -52,6 +52,12 @@ class TimeoutError(Exception):
     """
 
 
+class InvalidStateError(BaseException):
```

Comment:
The name here sounds a bit too general / risky to have it start being used by other things if all it currently means is e.g. `UnboxingWithoutResult`